data-structures/stack/stack.go

```diff
@@ -0,0 +1,33 @@
+// Package stack implements the Stack data structure, LIFO.
+package stack
+
+// Stack is the type that implements the Stack data structure.
+type Stack struct {
```

Comment:
Why enclose it in a struct?
You could try something like `type Stack []interface{}`